Automattic/Gravatar-SDK-Android
# QuickEditor: Avatar selection

Closes #224 

### Description

This PR adds the option to select one of the uploaded avatars. 

As per the design, when the avatar is clicked:
- an overlay is shown with the circular progress 
- avatar is instantly selected (reverts back if request fails)

We can't use `Scaffold` on the bottom sheet so I did add `SnackbarHost` manually. 

There's one known issue now https://github.com/Automattic/Gravatar-SDK-Android/issues/258
The avatar in the `ProfileCard` is not updated instantly - this is partially because there's no trigger for it yet in the ViewModel. Still, even if we refresh the profile the avatar won't update because of the cache.
This must be solved by loading the selected image manually or using the cache buster. 

I did update the DemoApp to update the avatar as well.


https://github.com/user-attachments/assets/335c59cb-e602-4c46-8a7b-5692e42000dc



### Testing Steps
**Important Note**: The backend is currently returning the wrong format for the `updated date` field. As a workaround, you can apply this patch to comment that field from the Avatar object.
[Gravatar-Android-13-04-48.patch](https://github.com/user-attachments/files/16455284/Gravatar-Android-13-04-48.patch)

1. Make sure you have at least two avatars uploaded on Gravatar
2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap `Update Avatar` button
5. Follow the steps to log in 
8. Select a different avatar
9. Confirm you can see the progress indicator
10. Confirm the new avatar was highlighted 
11. Confirm the snack was shown
